doc/string/end_with_p.rdoc

```diff
@@ -0,0 +1,17 @@
+Returns whether +self+ ends with any of the given +string_or_regexp+.
+
+Matches patterns against the end of +self+.
+For each given +string_or_regexp+, the pattern is:
+
+- +string_or_regexp+ itself, if it is a Regexp.
+- <tt>Regexp.quote(string_or_regexp)</tt>, if +string_or_regexp+ is a string.
```

Comment:
I don't think the pattern can be a regex:

```ruby
irb(main):019:0> 'hello'.end_with?(/\d+/)
(irb):19:in `end_with?': no implicit conversion of Regexp into String (TypeError)
```

Interestingly, `String#start_with?` can be a regex:

```ruby
irb(main):021:0> 'hello'.start_with?(/hell/)
=> true
```

Reply:
Fixed, here and in call-seq.